src/Reflection/ReflectionParameter.php

```diff
@@ -497,6 +494,31 @@ private function getClassFromNamedType(ReflectionNamedType $namedType): ?Reflect
         }
     }
 
+    private function detectIsOptional(): bool
```

Comment:
This seems to be improved here, but not really tested 🤔

Reply:
Rebased :)

Reply:
Indeed, solved in #847